spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/reload/ConfigReloadAutoConfiguration.java

```diff
@@ -71,32 +72,31 @@
 		@Autowired
 		private KubernetesClient kubernetesClient;
 
-		@Autowired
-		private ConfigMapPropertySourceLocator configMapPropertySourceLocator;
-
-		@Autowired
-		private SecretsPropertySourceLocator secretsPropertySourceLocator;
-
 		/**
 		 * @param properties config reload properties
 		 * @param strategy configuration update strategy
 		 * @return a bean that listen to configuration changes and fire a reload.
 		 */
 		@Bean
 		@ConditionalOnMissingBean
+		@ConditionalOnBean({ ConfigMapPropertySourceLocator.class,
+				SecretsPropertySourceLocator.class })
 		public ConfigurationChangeDetector propertyChangeWatcher(
-				ConfigReloadProperties properties, ConfigurationUpdateStrategy strategy) {
+				ConfigReloadProperties properties, ConfigurationUpdateStrategy strategy,
```

Comment:
No need for the `@Autowired`

Reply:
Without making that optional, it wont work for the configs i posted originally. Since you would potentially only have one or the other or both or none.